Usage of `cfg(any(docsrs, ...))` breaks downstream doc builds

I am currently using `swc_core@19.0.0` as a dependency (although this applies to later versions too).

I would like to use a `--cfg docsrs` to configure my own `doc_cfg`. However, this causes the doc build to break.

Full Command:
```sh
cargo +nightly doc --workspace --all-features --config "build.rustdocflags = ['--cfg', 'docsrs']" -vv
```

This seems to be caused by `docsrs` being included in `cfg(any(docsrs, ...))`.

It seems to have been mostly fixed in https://github.com/swc-project/swc/commit/9d793f1cd3f2e23ba5b286783e0633a13690f1c2, but a few instances remain in `swc_core/lib.rs` and `swc_core/plugin.rs`.

If you would like, I can look into fixing this.

<details>
<summary>Error</summary>

```rs
error[E0432]: unresolved import `swc_plugin_macro`
  --> C:\Users\Redfire\.cargo\registry\src\index.crates.io-1949cf8c6b5b557f\swc_core-19.0.0\src\plugin.rs:14:9
   |
14 | pub use swc_plugin_macro::css_plugin_transform;
   |         ^^^^^^^^^^^^^^^^ use of unresolved module or unlinked crate `swc_plugin_macro`
   |
   = help: if you wanted to use a crate named `swc_plugin_macro`, use `cargo add swc_plugin_macro` to add it to your `Cargo.toml`

error[E0432]: unresolved import `swc_plugin_macro`
  --> C:\Users\Redfire\.cargo\registry\src\index.crates.io-1949cf8c6b5b557f\swc_core-19.0.0\src\plugin.rs:29:9
   |
29 | pub use swc_plugin_macro::plugin_transform;
   |         ^^^^^^^^^^^^^^^^ use of unresolved module or unlinked crate `swc_plugin_macro`
   |
   = help: if you wanted to use a crate named `swc_plugin_macro`, use `cargo add swc_plugin_macro` to add it to your `Cargo.toml`

error[E0432]: unresolved import `swc_plugin_proxy`
  --> C:\Users\Redfire\.cargo\registry\src\index.crates.io-1949cf8c6b5b557f\swc_core-19.0.0\src\plugin.rs:76:13
   |
76 |     pub use swc_plugin_proxy::TransformPluginProgramMetadata;
   |             ^^^^^^^^^^^^^^^^ use of unresolved module or unlinked crate `swc_plugin_proxy`
   |
   = help: if you wanted to use a crate named `swc_plugin_proxy`, use `cargo add swc_plugin_proxy` to add it to your `Cargo.toml`

error[E0432]: unresolved import `swc_plugin_proxy`
  --> C:\Users\Redfire\.cargo\registry\src\index.crates.io-1949cf8c6b5b557f\swc_core-19.0.0\src\plugin.rs:94:13
   |
94 |     pub use swc_plugin_proxy::*;
   |             ^^^^^^^^^^^^^^^^ use of unresolved module or unlinked crate `swc_plugin_proxy`
   |
   = help: if you wanted to use a crate named `swc_plugin_proxy`, use `cargo add swc_plugin_proxy` to add it to your `Cargo.toml`

For more information about this error, try `rustc --explain E0432`.
error: could not document `swc_core`
```

</details>

--- a/crates/swc_core/src/lib.rs
+++ b/crates/swc_core/src/lib.rs
@@ -19,7 +19,13 @@ pub extern crate swc_ecma_quote_macros;
 
 // Plugins
 #[cfg(any(
-    docsrs,
+    all(
+        docsrs,
+        any(
+            feature = "__common_plugin_transform",
+            feature = "__plugin_transform_host"
+        )
+    ),
     feature = "__common_plugin_transform",
     feature = "__plugin_transform_host"
 ))]

--- a/crates/swc_core/src/plugin.rs
+++ b/crates/swc_core/src/plugin.rs
@@ -1,6 +1,12 @@
 // #[plugin_transform] macro
 #[cfg(any(
-    docsrs,
+    all(
+        docsrs,
+        any(
+            feature = "__common_plugin_transform",
+            feature = "__css_plugin_transform"
+        )
+    ),
     feature = "__common_plugin_transform",
     feature = "__css_plugin_transform",
 ))]
@@ -13,7 +19,14 @@
 )]
 pub use swc_plugin_macro::css_plugin_transform;
 #[cfg(any(
-    docsrs,
+    all(
+        docsrs,
+        any(
+            feature = "__common_plugin_transform",
+            feature = "__css_plugin_transform",
+            feature = "__ecma_plugin_transform"
+        )
+    ),
     feature = "__common_plugin_transform",
     feature = "__css_plugin_transform",
     feature = "__ecma_plugin_transform"
@@ -39,7 +52,14 @@ pub mod memory {
 /// Global HANDLER implementation for the plugin
 /// for error reporting.
 #[cfg(any(
-    docsrs,
+    all(
+        docsrs,
+        any(
+            feature = "__common_plugin_transform",
+            feature = "__css_plugin_transform",
+            feature = "__ecma_plugin_transform"
+        )
+    ),
     feature = "__common_plugin_transform",
     feature = "__css_plugin_transform",
     feature = "__ecma_plugin_transform"
@@ -58,7 +78,14 @@ pub mod errors {
 
 /// Plugin's environment metadata context.
 #[cfg(any(
-    docsrs,
+    all(
+        docsrs,
+        any(
+            feature = "__common_plugin_transform",
+            feature = "__css_plugin_transform",
+            feature = "__ecma_plugin_transform"
+        )
+    ),
     feature = "__common_plugin_transform",
     feature = "__css_plugin_transform",
     feature = "__ecma_plugin_transform"
@@ -79,7 +106,13 @@ pub mod metadata {
 /// Proxy to the host's data not attached to the AST, like sourcemap / comments.
 /// Or interfaces to setup the plugin's environment from the host.
 #[cfg(any(
-    docsrs,
+    all(
+        docsrs,
+        any(
+            feature = "__common_plugin_transform",
+            feature = "__plugin_transform_host"
+        )
+    ),
     feature = "__common_plugin_transform",
     feature = "__plugin_transform_host"
 ))]

